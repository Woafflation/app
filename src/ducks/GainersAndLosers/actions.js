import { getTimeRangeByDuration } from '../../utils/utils'

export const FETCH_GAINERS_LOSERS = '[gainersLosers] GAINERS_FETCH'
export const FETCH_GAINERS_LOSERS_SUCCESS =
  '[gainersLosers] GAINERS_FETCH_SUCCESS'
export const FETCH_GAINERS_LOSERS_FAILED =
  '[gainersLosers] GAINERS_FETCH_FAILED'
export const FETCH_GAINERS_LOSERS_CANCEL =
  '[gainersLosers] GAINERS_FETCH_CANCEL'
export const CLEAN_GAINERS_LOSERS = '[gainersLosers] GAINERS_CLEAN'

export const fetchGainersOrLosers = ({
  status = 'ALL',
  timeWindow = '15d',
  size = 10000
}) => ({
  type: FETCH_GAINERS_LOSERS,
  payload: {
    status,
    size,
    ...getTimeRangeByDuration(timeWindow)
  }
})
