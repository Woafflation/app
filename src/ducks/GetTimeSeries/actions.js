// timeseries
export const TIMESERIES_FETCH = '[timeseries] FETCH'
export const TIMESERIES_FETCH_SUCCESS = '[timeseries] FETCH_SUCCESS'
export const TIMESERIES_FETCH_FAILED = '[timeseries] FETCH_FAILED'

export const fetchTimeseries = settings => ({
  type: TIMESERIES_FETCH,
  payload: { ...settings }
})
