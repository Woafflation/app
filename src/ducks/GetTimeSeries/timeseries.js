import { DEV_ACTIVITY_QUERY } from './queries/dev_activity_query'
import { MARKET_SEGMENT_QUERY } from './queries/market_segment_query'
import { ETH_SPENT_OVER_TIME_QUERY } from './queries/eth_spent_over_time_query'
import { ETH_SPENT_OVER_TIME_BY_ALL_PROJECTS_QUERY } from './queries/eth_spent_over_time_by_all_projects_query'
import { BURN_RATE_QUERY } from './queries/burn_rate_query'
import { HISTORY_TWITTER_DATA_QUERY } from './queries/history_twitter_data_query'
import { SOCIAL_VOLUME_QUERY } from './queries/social_volume_query'
import { DAILY_ACTIVE_DEPOSITS_QUERY } from './queries/daily_active_deposits_query'
import { SOCIAL_DOMINANCE_QUERY } from './queries/social_dominance_query'
import { PERCENT_OF_TOKEN_SUPPLY_ON_EXCHANGES } from './queries/percent_of_token_supply_on_exchanges_query'
import { TOP_HOLDERS_PERCENT_OF_TOTAL_SUPPLY } from './queries/top_holders_percent_of_total_supply'
import { GAS_USED_QUERY } from './queries/gas_used'
import { PROJECT_TREND_HISTORY_QUERY } from './queries/project_trend_history_query'
import { METRIC_ANOMALIE_QUERY } from './queries/metric_anomaly_query'
import { GET_METRIC } from './queries/get_metric'
import { mergeTimeseriesByKey } from './../../utils/utils'
import { HISTORICAL_BALANCE_QUERY } from '../HistoricalBalance/common/queries'

const getMetricPreTransform = ({ getMetric: { timeseriesData } }) =>
  timeseriesData

const queryBuild = (type, preTransform = getMetricPreTransform) => {
  return {
    query: GET_METRIC(type),
    preTransform
  }
}
/*
   getMetric API available metrics could be seen by querying data with "getAvailableMetrics"
*/
const TIMESERIES = {
  ethSpentOverTime: {
    query: ETH_SPENT_OVER_TIME_QUERY,
    preTransform: ({ ethSpentOverTime: { ethSpentOverTime } }) =>
      ethSpentOverTime
  },
  EthSpentOverTimeByAllProjects: {
    query: ETH_SPENT_OVER_TIME_BY_ALL_PROJECTS_QUERY,
    preTransform: ({ ethSpentOverTimeByAllProjects: ethSpentOverTime }) =>
      ethSpentOverTime
  },
  nvt: {
    query: GET_METRIC('nvt'),
    preTransform: getMetricPreTransform
  },
  nvt_transaction_volume: {
    query: GET_METRIC('nvt_transaction_volume'),
    preTransform: getMetricPreTransform
  },
  burnRate: {
    query: BURN_RATE_QUERY
  },
  realized_value_usd: {
    query: GET_METRIC('realized_value_usd'),
    preTransform: getMetricPreTransform
  },
  historyTwitterData: {
    query: HISTORY_TWITTER_DATA_QUERY
  },
  historicalBalance: {
    query: HISTORICAL_BALANCE_QUERY
  },
  price_usd: {
    query: GET_METRIC('price_usd'),
    preTransform: getMetricPreTransform
  },
  volume_usd: {
    query: GET_METRIC('volume_usd'),
    preTransform: getMetricPreTransform
  },
  marketcap_usd: {
    query: GET_METRIC('marketcap_usd'),
    preTransform: getMetricPreTransform
  },
  devActivity: {
    query: DEV_ACTIVITY_QUERY
  },
  dev_activity: {
    query: GET_METRIC('dev_activity'),
    preTransform: getMetricPreTransform
  },
  socialDominance: {
    query: SOCIAL_DOMINANCE_QUERY
  },
  percentOfTokenSupplyOnExchanges: {
    query: PERCENT_OF_TOKEN_SUPPLY_ON_EXCHANGES,
    preTransform: ({ percentOnExchanges }) => percentOnExchanges
  },
  topHoldersPercentOfTotalSupply: {
    query: TOP_HOLDERS_PERCENT_OF_TOTAL_SUPPLY
  },
  age_destroyed: {
    query: GET_METRIC('age_destroyed'),
    preTransform: getMetricPreTransform
  },
  exchange_balance: {
    query: GET_METRIC('exchange_balance'),
    preTransform: getMetricPreTransform
  },
  daily_active_addresses: {
    query: GET_METRIC('daily_active_addresses'),
    preTransform: getMetricPreTransform
  },
  circulation: {
    query: GET_METRIC('circulation'),
    preTransform: getMetricPreTransform
  },
  dormant_circulation: {
    query: GET_METRIC('dormant_circulation'),
    preTransform: getMetricPreTransform
  },
  mvrv_usd: {
    query: GET_METRIC('mvrv_usd'),
    preTransform: getMetricPreTransform
  },
  dailyActiveDeposits: {
    query: DAILY_ACTIVE_DEPOSITS_QUERY
  },
  velocity: {
    query: GET_METRIC('velocity'),
    preTransform: getMetricPreTransform
  },
  transaction_volume: {
    query: GET_METRIC('transaction_volume'),
    preTransform: getMetricPreTransform
  },
  network_growth: {
    query: GET_METRIC('network_growth'),
    preTransform: getMetricPreTransform
  },
  social_volume_total: queryBuild('social_volume_total'),
  gasUsed: {
    query: GAS_USED_QUERY
  },
  socialVolume: {
    query: SOCIAL_VOLUME_QUERY,
    title: 'Social Volume',
    preTransform: data =>
      mergeTimeseriesByKey({
        key: 'datetime',
        timeseries: Object.values(data),
        mergeData: (longestTSData, timeserieData) => ({
          socialVolume: longestTSData.socialVolume + timeserieData.socialVolume,
          datetime: longestTSData.datetime
        })
      })
  },
  trendPositionHistory: {
    query: PROJECT_TREND_HISTORY_QUERY,
    preTransform: ({ getProjectTrendingHistory: data }) =>
      data.filter(({ position }) => position)
  },
  anomalies: {
    query: METRIC_ANOMALIE_QUERY,
    preTransform: ({ metricAnomaly: anomalies = [] }, metricAnomalyKey) => {
      return anomalies.map(anomaly => ({ ...anomaly, metricAnomalyKey })) || []
    }
  },
  marketSegment: {
    query: MARKET_SEGMENT_QUERY,
    preTransform: ({ devActivity }) => devActivity
  },
  mean_dollar_invested_age: {
    query: GET_METRIC('mean_dollar_invested_age'),
    preTransform: getMetricPreTransform
  },
  price_btc: queryBuild('price_btc'),
  price_eth: queryBuild('price_eth'),
  exchange_inflow: queryBuild('exchange_inflow'),
  exchange_outflow: queryBuild('exchange_outflow'),
  github_activity: queryBuild('github_activity'),
  community_messages_count_discord: queryBuild(
    'community_messages_count_discord'
  ),
  community_messages_count_telegram: queryBuild(
    'community_messages_count_telegram'
  ),
  community_messages_count_total: queryBuild('community_messages_count_total'),
  social_dominance_discord: queryBuild('social_dominance_discord'),
  social_dominance_professional_traders_chat: queryBuild(
    'social_dominance_professional_traders_chat'
  ),
  social_dominance_reddit: queryBuild('social_dominance_reddit'),
  social_dominance_total: queryBuild('social_dominance_total'),
  social_dominance_telegram: queryBuild('social_dominance_telegram'),
  social_volume_professional_traders_chat: queryBuild(
    'social_volume_professional_traders_chat'
  ),
  social_volume_reddit: queryBuild('social_volume_reddit'),
  social_volume_discord: queryBuild('social_volume_discord'),
  social_volume_telegram: queryBuild('social_volume_telegram'),
  social_volume_twitter: queryBuild('social_volume_twitter'),
  mean_age: queryBuild('mean_age'),

  mvrv_usd_intraday: queryBuild('mvrv_usd_intraday'),
  stock_to_flow: queryBuild('stock_to_flow')
}

export const hasMetric = metric => !!TIMESERIES[metric]
export const getMetricQUERY = metric => TIMESERIES[metric].query
export const getPreTransform = (name, metricAnomalyKey, alias = name) => {
  const transform = TIMESERIES[name].preTransform

  return ({ data, ...rest }) => ({
    ...rest,
    data: transform ? { [alias]: transform(data, metricAnomalyKey) } : data,
    __metric: alias
  })
}
