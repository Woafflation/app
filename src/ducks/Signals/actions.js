export const SIGNAL_CREATE = '[signal] CREATE_SIGNAL'
export const SIGNAL_CREATE_SUCCESS = '[signal] CREATE_SIGNAL_SUCCESS'
export const SIGNAL_CREATE_FAILED = '[signal] CREATE_SIGNAL_FAILED'
export const SIGNAL_UPDATE = '[signal] UPDATE_SIGNAL'
export const SIGNAL_UPDATE_SUCCESS = '[signal] UPDATE_SIGNAL_SUCCESS'
export const SIGNAL_UPDATE_FAILED = '[signal] UPDATE_SIGNAL_FAILED'
export const SIGNAL_FETCH_ALL = '[signal] FETCH_ALL'
export const SIGNAL_FETCH_ALL_SUCCESS = '[signal] FETCH_ALL_SUCCESS'
export const SIGNAL_FETCH_ALL_ERROR = '[signal] FETCH_ALL_ERROR'
// update signal
export const SIGNAL_TOGGLE_BY_ID = '[signal] TOGGLE_BY_ID'
export const SIGNAL_TOGGLE_SUCCESS = '[signal] TOGGLE_BY_ID_SUCCESS'
export const SIGNAL_TOGGLE_FAILED = '[signal] TOGGLE_BY_ID_FAILED'
// preview of history signal points
export const SIGNAL_FETCH_HISTORY_POINTS = '[signal] FETCH_HISTORY_POINTS'
export const SIGNAL_FETCH_HISTORY_POINTS_SUCCESS =
  '[signal] FETCH_HISTORY_POINTS_SUCCESS'
export const SIGNAL_FETCH_HISTORY_POINTS_FAILED =
  '[signal] FETCH_HISTORY_POINTS_FAILED'

export const WithoutChannelsError =
  'You must setup at least one channel for new signal'

export const createTrigger = ({
  target,
  metric,
  channels,
  timeWindow,
  title,
  description,
  cooldown,
  option,
  values = {
    percentThreshold: null
  }
}) => {
  const { percentThreshold } = values
  if (!channels || channels.length < 1) {
    throw new Error(WithoutChannelsError)
  }
  return {
    type: SIGNAL_CREATE,
    payload: {
      settings: {
        target,
        time_window: timeWindow,
        percent_threshold: percentThreshold,
        channel: channels[0].toLowerCase(),
        type: option
      },
      isPublic: false,
      title,
      description,
      cooldown
    }
  }
}

export const toggleTrigger = ({ id, isActive }) => ({
  type: SIGNAL_TOGGLE_BY_ID,
  payload: {
    id,
    active: !isActive
  }
})

export const fetchHistorySignalPoints = ({ cooldown, settings }) => ({
  type: SIGNAL_FETCH_HISTORY_POINTS,
  payload: {
    settings,
    cooldown
  }
})
