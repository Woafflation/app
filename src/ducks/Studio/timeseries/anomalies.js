import gql from 'graphql-tag'
import { Metrics, tooltipSettings } from '../../SANCharts/data'
import { persimmon } from '@santiment-network/ui/variables.scss'

export const OLD_ANOMALY_QUERY = gql`
  query metricAnomaly(
    $from: DateTime!
    $metric: AnomaliesMetricsEnum!
    $slug: String!
    $to: DateTime!
  ) {
    metricAnomaly(from: $from, to: $to, slug: $slug, metric: $metric) {
      datetime
    }
  }
`

export const GET_ANOMALY_QUERY = gql`
  query getAnomaly(
    $metric: String!
    $slug: String!
    $from: DateTime!
    $to: DateTime!
    $interval: interval
  ) {
    getAnomaly(anomaly: $metric) {
      timeseriesData(slug: $slug, from: $from, to: $to, interval: $interval) {
        datetime
      }
    }
  }
`

export const ANOMALIES = [
  'daily_active_addresses',
  'dev_activity',
  'exchange_balance'
]

export const AnomalyFetcher = {
  query: GET_ANOMALY_QUERY,
  preTransform: key => ({
    data: {
      getAnomaly: { timeseriesData }
    }
  }) =>
    timeseriesData.map(({ datetime }) => {
      const metric = key.replace('_anomaly', '')
      const { label: value } = Metrics[metric]
      return {
        key,
        metric,
        value,
        color: persimmon,
        datetime: +new Date(datetime)
      }
    })
}

export const OldAnomalyFetcher = {
  query: OLD_ANOMALY_QUERY,
  preTransform: metric => ({ data: { metricAnomaly } }) =>
    metricAnomaly.map(({ datetime }) => {
      const { label: value } = Metrics[metric]
      return {
        key: 'isAnomaly',
        color: persimmon,
        datetime: +new Date(datetime),
        metric,
        value
      }
    })
}

ANOMALIES.forEach(anomaly => {
  tooltipSettings[anomaly + '_anomaly'] = {
    label: 'Anomaly',
    formatter: v => v
  }
})

export function buildAnomalies (metrics) {
  return metrics
    .filter(({ key, anomalyKey }) => anomalyKey || ANOMALIES.includes(key))
    .map(({ key, anomalyKey }) => ({
      key: anomalyKey || key + '_anomaly',
      queryKey: anomalyKey ? 'anomalies' : 'anomaly',
      metricAnomaly: key
    }))
}
