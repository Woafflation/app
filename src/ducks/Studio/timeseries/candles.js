import { useMemo } from 'react'
import { Node } from '../../Chart/nodes'
import { PRICE_OHLC_QUERY, newOhlcPreTransformer } from './queries/price_ohlc'

export function useMetricSettingsAdjuster (MetricSettingMap) {
  return useMemo(
    () => {
      MetricSettingMap.forEach((MetricSettings, metric) => {
        if (MetricSettings.node !== Node.CANDLES) {
          delete MetricSettings.query
          delete MetricSettings.preTransform
          return
        }

        MetricSettings.query = PRICE_OHLC_QUERY
        MetricSettings.preTransform = newOhlcPreTransformer(metric)
      })

      return MetricSettingMap
    },
    [MetricSettingMap]
  )
}
