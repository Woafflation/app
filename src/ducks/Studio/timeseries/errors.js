const SUBSCRIPTION_INTERVAL_ANCHOR = 'outside the allowed interval'
const SUBSCRIPTION_INTERVAL_MSG =
  'Requested dates are outside of the allowed interval for your subscription'

export function substituteErrorMsg(backendMsg) {
  if (backendMsg.includes(SUBSCRIPTION_INTERVAL_ANCHOR)) {
    return SUBSCRIPTION_INTERVAL_MSG
  }

  return backendMsg
}
