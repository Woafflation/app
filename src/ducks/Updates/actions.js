// timeseries
export const NEW_APP_AVAILABLE = '[version] NEW_APP_AVAILABLE'

export const newAppAvailable = () => ({
  type: NEW_APP_AVAILABLE
})
