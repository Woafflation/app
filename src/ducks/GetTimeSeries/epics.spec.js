/* eslint-env jest */
import { ApolloClient } from 'apollo-client'
import { mockSingleLink } from 'react-apollo/test-utils'
import { ApolloLink, Observable } from 'apollo-link'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { ActionsObservable } from 'redux-observable'
import configureStore from 'redux-mock-store'
import fetchTimeseriesEpic from './epics'
import { getMetricQUERY } from './timeseries'
import * as actions from './actions'
const mockStore = configureStore([])

const createClient = link => {
  return new ApolloClient({
    link,
    cache: new InMemoryCache({
      addTypename: false
    })
  })
}

const mockedData = {
  historyPrice: [0, 1, 2, 3, 4, 5, 6, 7, 8].map(index => ({
    priceBtc: Math.round(100),
    priceUsd: Math.round(100),
    volume: Math.round(100),
    datetime: '2018-12-01T13:37:02.07080' + index,
    marketcap: Math.round(100)
  }))
}

const mockedDevActivity = {
  devActivity: [0, 1, 2, 3, 4, 5, 6, 7, 8].map(index => ({
    activity: Math.round(100),
    datetime: '2018-12-01T13:37:02.07080' + index
  }))
}

const link = mockSingleLink(
  {
    request: {
      query: getMetricQUERY('historyPrice'),
      variables: {
        slug: 'santiment',
        interval: '1d',
        from: '2018-12-01',
        to: '2018-12-10'
      }
    },
    result: { data: mockedData }
  },
  {
    request: {
      query: getMetricQUERY('historyPrice'),
      variables: {
        slug: 'santiment',
        interval: '1d',
        from: '2018-12-01',
        to: '2018-12-10'
      }
    },
    result: { data: mockedData }
  },
  {
    request: {
      query: getMetricQUERY('devActivity'),
      variables: {
        slug: 'santiment',
        interval: '1d',
        from: '2018-12-01',
        to: '2018-12-10',
        transform: 'movingAverage',
        movingAverageIntervalBase: 7
      }
    },
    result: { data: mockedDevActivity }
  },
  {
    request: {
      query: getMetricQUERY('historyPrice'),
      variables: {
        slug: 'santiment',
        interval: '1d',
        from: '2018-12-01',
        to: '2018-12-10'
      }
    },
    result: { data: mockedData }
  },
  {
    request: {
      query: getMetricQUERY('devActivity'),
      variables: {
        slug: 'santiment',
        interval: '1d',
        from: '2018-12-01',
        to: '2018-12-10',
        transform: 'movingAverage',
        movingAverageIntervalBase: 7
      }
    },
    result: { data: mockedDevActivity }
  }
)

describe('Fetch timeseries', () => {
  it('should return price succesfull', async () => {
    const client = await createClient(link)
    const action$ = ActionsObservable.of({
      type: actions.TIMESERIES_FETCH,
      payload: {
        id: 0,
        historyPrice: {
          from: '2018-12-01',
          to: '2018-12-10',
          slug: 'santiment',
          interval: '1d'
        }
      }
    })
    const epic$ = fetchTimeseriesEpic(action$, mockStore({}), { client })
    const promise = epic$.toArray().toPromise()
    const result = await promise
    expect(result[0].type).toEqual(actions.TIMESERIES_FETCH_SUCCESS)
  })

  it('should return price + devActivity succesfull', async () => {
    const client = await createClient(link)
    const action$ = ActionsObservable.of({
      type: actions.TIMESERIES_FETCH,
      payload: {
        id: 0,
        historyPrice: {
          from: '2018-12-01',
          to: '2018-12-10',
          slug: 'santiment',
          interval: '1d'
        },
        devActivity: {
          from: '2018-12-01',
          to: '2018-12-10',
          slug: 'santiment',
          interval: '1d',
          transform: 'movingAverage',
          movingAverageIntervalBase: 7
        }
      }
    })
    const epic$ = fetchTimeseriesEpic(action$, mockStore({}), { client })
    const promise = epic$.toArray().toPromise()
    const result = await promise
    expect(result[0].type).toEqual(actions.TIMESERIES_FETCH_SUCCESS)
    expect(result).toMatchSnapshot()
  })

  it('should return price + devActivity as one merged timeseries array by datetime', async () => {
    const client = await createClient(link)
    const action$ = ActionsObservable.of({
      type: actions.TIMESERIES_FETCH,
      payload: {
        id: 0,
        historyPrice: {
          from: '2018-12-01',
          to: '2018-12-10',
          slug: 'santiment',
          interval: '1d'
        },
        devActivity: {
          from: '2018-12-01',
          to: '2018-12-10',
          slug: 'santiment',
          interval: '1d',
          transform: 'movingAverage',
          movingAverageIntervalBase: 7
        },
        meta: {
          mergedByDatetime: true
        }
      }
    })
    const epic$ = fetchTimeseriesEpic(action$, mockStore({}), { client })
    const promise = epic$.toArray().toPromise()
    const result = await promise
    expect(result[0].type).toEqual(actions.TIMESERIES_FETCH_SUCCESS)
    expect(result).toMatchSnapshot()
  })

  it('should return error, if metric doesnt exist', async () => {
    const client = await createClient(link)
    const action$ = ActionsObservable.of({
      type: actions.TIMESERIES_FETCH,
      payload: {
        id: 0,
        anyStrangeMetric: {
          from: '2018-12-01',
          to: '2018-12-10',
          slug: 'santiment',
          interval: '1d'
        }
      }
    })
    try {
      const epic$ = fetchTimeseriesEpic(action$, mockStore({}), { client })
      await epic$.toArray().toPromise()
    } catch (e) {
      expect(e).toBeDefined()
      expect(e).toMatchSnapshot()
    }
  })
})
