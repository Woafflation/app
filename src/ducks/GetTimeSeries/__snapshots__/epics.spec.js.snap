// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch timeseries should return error, if metric doesnt exist 1`] = `
[Error: 
            Unsupported metric yet: "anyStrangeMetric".
            Add a query for this metric and description.
          ]
`;

exports[`Fetch timeseries should return price + devActivity as one merged timeseries array by datetime 1`] = `
Array [
  Object {
    "payload": Object {
      "0": Object {
        "devActivity": Object {
          "isEmpty": false,
          "isError": false,
          "isLoading": false,
        },
        "errorMetrics": Object {},
        "historyPrice": Object {
          "isEmpty": false,
          "isError": false,
          "isLoading": false,
        },
        "isLoading": false,
        "timeseries": Array [
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070800",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070801",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070802",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070803",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070804",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070805",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070806",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070807",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070808",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
        ],
      },
    },
    "type": "[timeseries] FETCH_SUCCESS",
  },
]
`;

exports[`Fetch timeseries should return price + devActivity succesfull 1`] = `
Array [
  Object {
    "payload": Object {
      "0": Object {
        "devActivity": Object {
          "isEmpty": false,
          "isError": false,
          "isLoading": false,
        },
        "errorMetrics": Object {},
        "historyPrice": Object {
          "isEmpty": false,
          "isError": false,
          "isLoading": false,
        },
        "isLoading": false,
        "timeseries": Array [
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070800",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070801",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070802",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070803",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070804",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070805",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070806",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070807",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
          Object {
            "activity": 100,
            "datetime": "2018-12-01T13:37:02.070808",
            "marketcap": 100,
            "priceBtc": 100,
            "priceUsd": 100,
            "volume": 100,
          },
        ],
      },
    },
    "type": "[timeseries] FETCH_SUCCESS",
  },
]
`;
